packages/components/modal/__tests__/modal.test.tsx

```diff
@@ -108,4 +124,74 @@ describe("Modal", () => {
     fireEvent.keyDown(modal, {key: "Escape"});
     expect(onClose).toHaveBeenCalledTimes(1);
   });
+
+  it("should be rendered a draggable modal", () => {
+    // mock viewport size to 1920x1080
+    jest.spyOn(document.documentElement, "clientWidth", "get").mockImplementation(() => 1920);
+    jest.spyOn(document.documentElement, "clientHeight", "get").mockImplementation(() => 1080);
+
+    const wrapper = render(<ModalDraggable />);
+
+    const modal = wrapper.getByRole("dialog");
+    const modalHeader = wrapper.getByText("Modal header");
+
+    fireEvent.touchStart(modalHeader, {changedTouches: [{pageX: 0, pageY: 0}]});
+    fireEvent.touchMove(modalHeader, {changedTouches: [{pageX: 100, pageY: 50}]});
+    fireEvent.touchEnd(modalHeader, {changedTouches: [{pageX: 100, pageY: 50}]});
+
+    expect(() => wrapper.unmount()).not.toThrow();
+    expect(document.documentElement.clientWidth).toBe(1920);
+    expect(document.documentElement.clientHeight).toBe(1080);
+    expect(modalHeader.style.cursor).toBe("move");
+    expect(modal.style.transform).toBe("translate(100px, 50px)");
+  });
+
+  it("should not drag overflow viewport", () => {
+    // mock viewport size to 1920x1080
+    jest.spyOn(document.documentElement, "clientWidth", "get").mockImplementation(() => 1920);
+    jest.spyOn(document.documentElement, "clientHeight", "get").mockImplementation(() => 1080);
+    const wrapper = render(<ModalDraggable />);
+    const modal = wrapper.getByRole("dialog");
+    const modalHeader = wrapper.getByText("Modal header");
+
+    fireEvent.touchStart(modalHeader, {changedTouches: [{pageX: 100, pageY: 50}]});
+    fireEvent.touchMove(modalHeader, {changedTouches: [{pageX: 10000, pageY: 5000}]});
+    fireEvent.touchEnd(modalHeader, {changedTouches: [{pageX: 10000, pageY: 5000}]});
+
+    expect(modal.style.transform).toBe("translate(1920px, 1080px)");
+  });
+
+  it("should not drag when disabled", () => {
+    // mock viewport size to 1920x1080
+    jest.spyOn(document.documentElement, "clientWidth", "get").mockImplementation(() => 1920);
+    jest.spyOn(document.documentElement, "clientHeight", "get").mockImplementation(() => 1080);
+    const wrapper = render(<ModalDraggable isDisabled />);
+    const modal = wrapper.getByRole("dialog");
+    const modalHeader = wrapper.getByText("Modal header");
+
+    fireEvent.touchStart(modalHeader, {changedTouches: [{pageX: 100, pageY: 50}]});
+    fireEvent.touchMove(modalHeader, {changedTouches: [{pageX: 200, pageY: 100}]});
+    fireEvent.touchEnd(modalHeader, {changedTouches: [{pageX: 200, pageY: 100}]});
+
+    expect(modal.style.transform).toBe("");
+  });
+
+  test("should be rendered a draggable modal with overflow", () => {
+    // mock viewport size to 1920x1080
```

Comment:
Since workaround code has been added, it would be helpful if you could add tests for mobile-sized devices as well!

Reply:
> Since workaround code has been added, it would be helpful if you could add tests for mobile-sized devices as well!

I have add a test for  mobile-sized devices.